src/udev/udev-spawn.c

```diff
@@ -73,15 +74,17 @@ static int on_spawn_io(sd_event_source *s, int fd, uint32_t revents, void *userd
                 return 0;
         }
 
+        l_orig = l;
         if ((size_t) l == size) {
```

Comment:
This condition needs to be updated, otherwise, the truncated flag is set by filling stderr.